src/Security/Certificate.cs

```diff
@@ -369,22 +440,63 @@ public NSData GetNormalizedSubjectSequence ()
 
 #if MONOMAC
 		[DllImport (Constants.SecurityLibrary)]
-		[Deprecated (PlatformName.MacOSX, 10,13)]
-		static extern /* __nullable CFDataRef */ IntPtr SecCertificateCopySerialNumber (IntPtr /* SecCertificateRef */ certificate, IntPtr /* CFErrorRef * */ error);
+#if !NET
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
 #else
+		[UnsupportedOSPlatform ("macos.10.13")]
+#if MONOMAC
+		[Obsolete ("Starting with macos.10.13 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#endif
+#endif
+		static extern /* __nullable CFDataRef */ IntPtr SecCertificateCopySerialNumber (IntPtr /* SecCertificateRef */ certificate, IntPtr /* CFErrorRef * */ error);
+#else // !MONOMAC
+#if !NET
 		[iOS (10,3)]
-		[Deprecated (PlatformName.iOS, 11,0)]
-		[Deprecated (PlatformName.MacOSX, 10,13)]
-		[Deprecated (PlatformName.WatchOS, 4,0)]
-		[Deprecated (PlatformName.TvOS, 11,0)]
+#else
+		[SupportedOSPlatform ("ios10.3")]
+#endif
+#if !NET
+		[Deprecated (PlatformName.iOS, 11, 0)]
+		[Deprecated (PlatformName.TvOS, 11, 0)] 
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
+		[Deprecated (PlatformName.WatchOS, 4, 0)]
+#else
+		[UnsupportedOSPlatform ("ios11.0")]
+		[UnsupportedOSPlatform ("tvos11.0")]
+		[UnsupportedOSPlatform ("macos10.13")]
+#if IOS
+		[Obsolete ("Starting with ios11.0 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+		[Obsolete ("Starting with tvos11.0 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif MONOMAC
+		[Obsolete ("Starting with macos10.13 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#endif
+#endif
 		[DllImport (Constants.SecurityLibrary)]
 		static extern /* __nullable CFDataRef */ IntPtr SecCertificateCopySerialNumber (IntPtr /* SecCertificateRef */ certificate);
 #endif
+#if !NET
 		[iOS (10,3)]
-		[Deprecated (PlatformName.iOS, 11,0, message: "Use 'GetSerialNumber(out NSError)' instead.")]
-		[Deprecated (PlatformName.MacOSX, 10,13, message: "Use 'GetSerialNumber(out NSError)' instead.")]
-		[Deprecated (PlatformName.WatchOS, 4,0, message: "Use 'GetSerialNumber(out NSError)' instead.")]
-		[Deprecated (PlatformName.TvOS, 11,0, message: "Use 'GetSerialNumber(out NSError)' instead.")]
```

Comment:
^ why re-order them ?

Reply:
I used the snippet and this is the order the snippet is in :/ haha I'll copy and paste the originals